test/suite/test_checkpoint_snapshot06.py

```diff
@@ -0,0 +1,250 @@
+        # extend under them. Until that's fixed, expect (not just allow) those rows to
+        # exist and demand that they read back as zero and not as check_value. When it
+        # is fixed (so the end of the table updates transactionally) the special-case
+        # logic can just be removed.
+        flcs_tolerance = more_invisible_rows_exist and self.value_format == '8t'
+
+        session = self.session
+        if read_ts == 0:
+            session.begin_transaction()
+        else:
+            session.begin_transaction('read_timestamp=' + self.timestamp_str(read_ts))
+        cursor = session.open_cursor(uri)
+        count = 0
+        for k, v in cursor:
+            if flcs_tolerance and count >= nrows:
+                self.assertEqual(v, 0)
+            else:
+                self.assertEqual(v, check_value)
+            count += 1
+        session.commit_transaction()
+        self.assertEqual(count, nrows * 2 if flcs_tolerance else nrows)
+
+    def perform_backup_or_crash_restart(self, fromdir, todir):
+        if self.restart == True:
+            #Simulate a crash by copying to a new directory(RESTART).
+            copy_wiredtiger_home(self, fromdir, todir + ".copy", True)
+            simulate_crash_restart(self, fromdir, todir)
+        else:
+            #Take a backup and restore it.
+            self.take_full_backup(fromdir, todir)
+            self.take_full_backup(fromdir, todir + ".copy")
+            self.reopen_conn(todir)
+
+    def test_checkpoint_snapshot(self):
+        self.moresetup()
+
+        ds_1 = SimpleDataSet(self, self.uri_1, 0, \
+                key_format=self.key_format, value_format=self.value_format, \
+                config="log=(enabled=true)" +self.extraconfig)
+        ds_1.populate()
+
+        ds_2 = SimpleDataSet(self, self.uri_2, 0, \
+                key_format=self.key_format, value_format=self.value_format, \
+                config="log=(enabled=true)" + self.extraconfig)
+        ds_2.populate()
+
+        # Insert number of records into both tables.
+        self.large_updates(self.uri_1, self.valuea, ds_1, self.nrows, 0)
+        self.check(self.valuea, self.uri_1, self.nrows, 0, False)
+
+        self.large_updates(self.uri_2, self.valuea, ds_2, self.nrows, 0)
+        self.check(self.valuea, self.uri_2, self.nrows, 0, False)
+
+        # Remove one key from both the tables.
+        cursor1 = self.session.open_cursor(self.uri_1)
+        cursor2 = self.session.open_cursor(self.uri_2)
+
+        cursor1.set_key(ds_1.key(50))
+        cursor2.set_key(ds_2.key(50))
+        self.assertEqual(cursor1.remove(), 0)
+        self.assertEqual(cursor2.remove(), 0)
+
+        # Truncate the range from 1 - 100 where the key 50 doesn't exist.
+        session1 = self.conn.open_session()
+        session1.begin_transaction()
+        cursor11 = session1.open_cursor(self.uri_1)
+        cursor12 = session1.open_cursor(self.uri_2)
+
+        cursor11.set_key(ds_1.key(100))
```

Comment:
Should this be key 1 instead of 100? Or at least the comment says that it should be. Ok, I see that we're only using a "stop" cursor. Please expand the comment to make that clear. Something like:
Truncate the range from 1-100 in both tables where key 50 doesn't exist.
We only set a stop cursor for both tables and send in an empty start cursor to truncate
from the beginning of the table.

Reply:
Fixed as suggested.